tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/DeLambdifier.scala

```diff
@@ -0,0 +1,29 @@
+package at.forsyte.apalache.tla.pp
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import at.forsyte.apalache.tla.lir.transformations._
+
+/**
+ * DeLambdifier turns applications of operator-typed LET-expressions, into LET-expressions with applications pushed into
+ * the body.
+ *
```

Comment:
Maybe note that this is a workaround motivated by lack of support for lambdas on the IR, with a link to the issue?